test/benchmark/post_layout_optimization.cpp

```diff
@@ -0,0 +1,65 @@
+//
+// Created by simon on 14.10.24.
+//
+
+#include <catch2/benchmark/catch_benchmark.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+#include "../utils/blueprints/network_blueprints.hpp"
+
+#include <fiction/algorithms/physical_design/orthogonal.hpp>
+#include <fiction/algorithms/physical_design/post_layout_optimization.hpp>
+#include <fiction/layouts/cartesian_layout.hpp>
+#include <fiction/layouts/clocked_layout.hpp>
+#include <fiction/layouts/coordinates.hpp>
+#include <fiction/layouts/gate_level_layout.hpp>
+#include <fiction/layouts/tile_based_layout.hpp>
+
+#include <mockturtle/networks/aig.hpp>
+
+using namespace fiction;
+
+TEST_CASE("Benchmark Post-Layout Optimization", "[benchmark]")
+{
+    using gate_layout = gate_level_layout<clocked_layout<tile_based_layout<cartesian_layout<offset::ucoord_t>>>>;
+
+    const auto                      ntk    = blueprints::parity_network<mockturtle::aig_network>();
+    const auto                      layout = orthogonal<gate_layout>(ntk);
+    post_layout_optimization_params params{};
+
+    BENCHMARK("post_layout_optimization: full optimization")
+    {
+        post_layout_optimization<gate_layout>(layout.clone());
+    };
+
+    params.max_gate_relocations = 0;
+
+    BENCHMARK("post_layout_optimization: wiring reduction only")
+    {
+        post_layout_optimization<gate_layout>(layout.clone(), params);
+    };
```

Comment:
Currently, the `clone()` call is also part of the runtime measurement. Is that intended behavior?

Reply:
As `post_layout_optimization` optimizes the layout directly, I need to pass the unoptimized layout. Furthermore, the computational overhead of cloning the layout should be negligible and also a constant time overhead.

Reply:
You could clone outside of the `BENCHMARK` call. It should be negligible, but up to you.

Reply:
That leads to the same problem, as then the cloned layout is optimized directly.